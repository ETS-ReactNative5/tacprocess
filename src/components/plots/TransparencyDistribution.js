import React from 'react'
import PropTypes from 'prop-types'

import Histogram from './Histogram'

/**
 *
 */
const TransparencyDistribution = ({ observingConditionsClouds }) => {
  const transparencies = ['Clear', 'Thin cloud', 'Thick Cloud', 'Any']

  const datasets = [
    {
      className: 'all-partners',
      data: {
        'Clear': observingConditionsClouds.timeRequested.clear,
        'Thin cloud': observingConditionsClouds.timeRequested.thinCloud,
        'Thick Cloud': observingConditionsClouds.timeRequested.thickCloud,
        'Any': observingConditionsClouds.timeRequested.any
      }
    }
  ]

  return <Histogram
    keys={ transparencies }
    datasets={ datasets }
    xLabel='Transparency'
    yLabel='Requested Time (hrs)'/>
}

TransparencyDistribution.propTypes = {
  observingConditionsClouds: PropTypes.object.isRequired
}

export default TransparencyDistribution
