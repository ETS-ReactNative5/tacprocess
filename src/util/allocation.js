import { isFloat } from "../util";


export function illegalAllocation(element, index, array){
  if(
    !(isFloat(element.allocatedTime.p0)) ||
    !(isFloat(element.allocatedTime.p1)) ||
    !(isFloat(element.allocatedTime.p2)) ||
    !(isFloat(element.allocatedTime.p3)) ||
    !(isFloat(element.allocatedTime.p4)) ||
    !(parseFloat(element.allocatedTime.p0) >= 0) ||
    !(parseFloat(element.allocatedTime.p1) >= 0) ||
    !(parseFloat(element.allocatedTime.p2) >= 0) ||
    !(parseFloat(element.allocatedTime.p3) >= 0) ||
    !(parseFloat(element.allocatedTime.p4) >= 0)
){
    return true
  }
  return false
}

export function checkAllocatedTimes(proposals){
  /*
   *
   *
   * @param proposals and array of proposals
   * @return true if all the allocated times correct else false
   */
  return !proposals.some(illegalAllocation)
}

export function getQuaryToAddAllocation(proposals, partner, semester){
  const pList = []
  if (checkAllocatedTimes(proposals)){

    proposals.forEach(p => (
      [0, 1, 2, 3, 4].forEach( t => {
        const priority = `p${t}`
        pList.push(
        `{
            proposalCode: "${p.proposalCode}",
            priority: ${t},
            time: ${p.allocatedTime[priority]}
         }`
      )}
    )
    ))
  }
  const mutateQuery = `
      mutation uta {
        updateTimeAllocations(timeAllocations: {
          partner: "${ partner }",
          semester: "${ semester }",
          timeAllocations: [
            ${pList}
          ]
        }) {
          success
        }
      }
  `
  return mutateQuery
}
