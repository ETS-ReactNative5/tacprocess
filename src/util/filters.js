import { ALL_PARTNER,
	HOME_PAGE,
	STATISTICS_PAGE,
	DOCUMENTATION_PAGE,
	TECHNICAL_PAGE,
	TAC_PAGE,
	ADMIN_PAGE
} from "../types"


export function totalTimeRequestedPerParner(proposals, semester, partner="All" ){
	/*
	*
	* @param proposals of a selected partner all or single
	* @param semester selected semester
	* @param partner all partners or a single
	* @return total
	*/
	
	let total = 0;
	proposals.forEach( p =>{
		p.timeRequests.forEach( r => {
			if (r.semester === semester){
				r.distribution.forEach(d => {
					if ( partner === "All"){
						total += d.time
					}else if (d.partnerCode === partner){
						total += d.time
					}
				})
				
			}
		})
	});
	return total
}

export const semestersArray = () => {
	let startYear = 2006;
	const today = new Date();
	const year = today.getFullYear();
	let semester = [];
	while (startYear < year + 8){
		semester.push(
			`${ startYear }-1`, `${ startYear }-2`
		);
		startYear += 1
	}
	return semester
};

export const firstSelectedPartner = roles => {
	let first = ALL_PARTNER;
	
	for (let r of roles || []) {
		if (r.type === "ADMINISTRATOR") {
			first = ALL_PARTNER;
			break;
		}
		if (r.type === "SALT_ASTRONOMER") {
			first = ALL_PARTNER;
			break;
		}
		if (r.type === "TAC_CHAIR") {
			first = r.partners[0] || "";
		}
		if (r.type === "TAC_MEMBER") {
			first = r.partners[0] || "";
		}
	}
	
	return first
};

export const getPartnerList = roles => {
	let partnerList = [];
	for (let r of roles || []) {
		if (r.type === "ADMINISTRATOR" || r.type === "SALT_ASTRONOMER") {
			partnerList = r.partners;
			partnerList.includes(ALL_PARTNER) ? partnerList.push() : partnerList.push(ALL_PARTNER);
			break;
		}
		
		if (r.type === "TAC_CHAIR") {
			partnerList = r.partners;
		}
		if (r.type === "TAC_MEMBER") {
			partnerList = r.partners;
		}
	}
	return partnerList
};

/*
* @params list an array of
*/
export const listForDropdown = list => {
	return (list || []).map( l => ({ label: l, value: l }))
};

/**
 * method convert a standard SA object to an array of SA names
 *
 * @params Array of SALT astronomers directly from server
 * @return Array of SALT astronomers names
 * */
export const getAstronomersList = saList => {
	return (saList || []).map( l => (`${l.name}`))
};

/**
 * get a current uri path and return the selected page name
 *
 * @params pathname uri path
 * @return name of selected page or Home page by default
 */
export const loadedPage = pathname => {
	console.log({pathname});
	return pathname === "/"? HOME_PAGE :
		pathname === "/timeallocation"? TAC_PAGE :
			pathname === "/statistics"? STATISTICS_PAGE :
				pathname === "/techreview"? TECHNICAL_PAGE :
					pathname === "/documentation"? DOCUMENTATION_PAGE :
						pathname === "/admin"? ADMIN_PAGE : HOME_PAGE;
};

/**\
 * It reduce the proposals to only proposals that are assigned to @param astronomer
 * if @param astronomer is "All" all are returned
 * if @param astronomer is "Assigned" only assigned proposals returned
 * if @param astronomer is "Not Assigned" only unassigned proposals returned
 * if @param astronomer is "SA name" only proposals that are assigned to "SA name" can be returned
 *
 * @param proposals array of proposals
 * @param astronomer a SALT astronomer
 * @return proposals reduced proposals
 * */

export const reduceProposalsPerAstronomer = (proposals, astronomer) => {
	let prop = [];
	if (astronomer === "All"){
		prop = proposals
	}
	else if (astronomer === "Assigned"){
		proposals.forEach(p => {
			if (p.liaisonAstronomer !== null) {prop.push(p)}
		})
	}
	else if (astronomer === "Not Assigned"){
		proposals.forEach(p => {
			if (p.liaisonAstronomer === null) {prop.push(p)}
		})
	}else {
		proposals.forEach(p => {
			if (p.liaisonAstronomer === astronomer) {prop.push(p)}
		})
	}
	
	return prop
};
