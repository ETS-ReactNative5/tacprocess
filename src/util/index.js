import * as types from '../types';

/**
 * Get the observing time for a semester in a proposal.
 * If a partner is given, the observing time is the time requested for that partner.
 *
 * @param proposal
 * @param semester
 * @param partner
 * @returns {*}
 */
export function proposalObservingTime(proposal, semester, partner='All') {
    return proposal.timeRequests
            .filter(r => r.semester === semester) // semester is correct
            .reduce((allDistributionItems, r) => [...allDistributionItems, ...r.distribution], []) // collect all partner time requests
            .filter(d => partner === 'All' || d.partnerCode === partner) // partner is correct
            .reduce((sum, d) => sum + d.time, 0); // add up all time requests
}
/**
 * Get the observing time requested for a semester and transparency in a list of proposals.
 * If a partner is given, the observing time is calculated for that partner only.
 *
 * @param proposals
 * @param semester
 * @param transparency
 * @param partner
 */
export function observingTimeForTransparency(proposals, semester, transparency, partner='All') {
    return proposals
            .filter(p => p.transparency === transparency) // transparency is correct
            .reduce((allTimeRequests, p) => [...allTimeRequests, ...p.timeRequests], []) // collect time requests
            .filter(r => r.semester === semester) // semester is correct
            .reduce((allDistributionItems, r) => [...allDistributionItems, ...r.distribution], []) // collect partner time requests
            .filter(d => partner === 'All' || d.partnerCode === partner) // partner is correct
            .reduce((sum, request) => sum + request.time, 0); // add up all time requests
}

/**
 * Get the observing time requested for a semester and seeing range in a list of proposals.
 * If a partner is given, the observing time is calculated for that partner only.
 *
 * The seeing range must be an array of the minimum and maximum value the proposal's maximum
 * seeing may have. The minimum value is inclusive, the maximum value exclusive.
 *
 * @param proposals
 * @param semester
 * @param seeingRange
 * @param partner
 */
export function observingTimeForSeeing(proposals, semester, seeingRange, partner='All') {
    return proposals
            .filter(p => seeingRange[0] <= p.maxSeeing && p.maxSeeing < seeingRange[1])// seeing is correct
            .reduce((allTimeRequests, p) => [...allTimeRequests, ...p.timeRequests], []) // collect time requests
            .filter(r => r.semester === semester) // semester is correct
            .reduce((allDistributionItems, r) => [...allDistributionItems, ...r.distribution], []) // collect partner time requests
            .filter(d => partner === 'All' || d.partnerCode === partner) // partner is correct
            .reduce((sum, request) => sum + request.time, 0); // add up all time requests
}

/**
 * Get the number of instrument configurations in a proposal.
 *
 * @param proposal
 * @returns {*}
 */
export function instrumentCount(proposal) {
    return Object.keys(proposal.instruments)
            .reduce((sum, key) => sum + proposal.instruments[key].length, 0);
}

/**
 * Get the observing time requested for a semester and instrument in a proposal.
 * If a partner is given, the observing time is calculated for that partner only. If a field is
 * given, in the JavaScript object representing the object this field must have the given
 * value.
 *
 * If a proposal has more than one instrument configuration, the time is split evenly between
 * the configurations.
 *
 * The instrument argument is case-insensitive.
 *
 * The partner, field and value must be passed as an object. For example,
 *
 * observingTimeForInstrument(proposals, '2017-1', 'rss', {partner: 'RSA'})
 * observingTimeForInstrument(proposals, '2017-1', 'rss', {partner: 'RSA', field: 'mode', value: 'Polarimetry'})

 * @param proposal
 * @param semester
 * @param instrument
 * @param field
 * @param value
 * @param partner
 */
export function proposalObservingTimeForInstrument(proposal, semester, instrument, {field, value, partner='All'}) {
    const instrumentModeCount = (proposal.instruments[instrument.toLowerCase()] || [])
            .filter(v => !field || v[field] === value)
            .length;

    const instrumentModeFraction = instrumentModeCount / instrumentCount(proposal);

    const totalObservingTime = proposal.timeRequests
            .filter(r => r.semester === semester) // semester is correct
            .reduce((distributionItems, r) => [...distributionItems, ...r.distribution], []) // collect all partner time requests
            .filter(d => partner === 'All' || d.partnerCode === partner) // partner is correct
            .reduce((sum, d) => sum + d.time, 0); // add all time requests

    return instrumentModeFraction * totalObservingTime;
}

/**
 * Get the observing time requested for a semester and instrument in a list of proposals.
 * If a partner is given, the observing time is calculated for that partner only. If a field is
 * given, in the JavaScript object representing the object this field must have the given
 * value.
 *
 * If a proposal has more than one instrument configuration, the time is split evenly between
 * the configurations.
 *
 * The instrument argument is case-insensitive.
 *
 * The partner, field and value must be passed as an object. For example,
 *
 * observingTimeForInstrument(proposals, '2017-1', 'rss', {partner: 'RSA'})
 * observingTimeForInstrument(proposals, '2017-1', 'rss', {partner: 'RSA', field: 'mode', value: 'Polarimetry'})

 * @param proposals
 * @param semester
 * @param instrument
 * @param field
 * @param value
 * @param partner
 */
export function observingTimeForInstrument(proposals, semester, instrument, {field, value, partner}) {
    return proposals
            .reduce((sum, proposal) =>
                            sum + proposalObservingTimeForInstrument(proposal,
                                                                     semester,
                                                                     instrument,
                                                                     {field, value, partner}), 0);
}

/**
 * Get the sorted list of partners a user should see.
 *
 * A partner is included in this list if it is included in any of the user's roles. If the user is an
 * administrator, the partner 'All' is added to the list as well.
 *
 * @param user
 */
export function partners(user) {
    if (!user.roles) {
        return [];
    }
    const initial = hasRole(user, types.ADMINISTRATOR) || hasRole(user, types.SALT_ASTRONOMER) ? [types.ALL_PARTNER] : [];
    const rolePartners = user.roles // collect partners from all roles
            .reduce((prev, role) => [...(role.partners || []), ...prev], initial);
    const partnerSet = new Set(rolePartners);

    return Array.from(partnerSet).sort();
}

/**
 * Check whether a user has a role for a partner.
 *
 * The partner is ignored for the roles of administrator and SALT Astronomer.
 *
 * The partner "All" is not supported.
 *
 * @param user
 * @param role
 * @param partner
 */
export function hasRole(user, role, partner) {
    if (role === types.ADMINISTRATOR || role === types.SALT_ASTRONOMER) {
        return (user.roles || []).some(r => r.role === role);
    } else {
        return (user.roles || []).some(r => r.role === role && (r.partners || []).includes(partner));
    }
}

export function canDo(user, action, partner) {
    switch (action) {
    case types.VIEW_TIME_ALLOCATION_PAGE:
        return hasRole(user, types.TAC_MEMBER, partner) ||
                hasRole(user, types.SALT_ASTRONOMER) ||
                hasRole(user, types.ADMINISTRATOR);
    case types.EDIT_TIME_ALLOCATION_PAGE:
        return hasRole(user, types.TAC_CHAIR, partner) ||
                hasRole(user, types.ADMINISTRATOR);
    default:
        return false;
    }
}

export function isFloat(val) {
    const floatRegex = /^[+-]?\d+(?:[.,]\d*?)?$/;
    if (!floatRegex.test(val))
        return false;

    const temp = parseFloat(val);
    if (isNaN(temp))
        return false;
    return true;
}

export function canUserWriteAllocations(user, partner){
  let canWrite = false;
  user.partners.forEach( p => {
    if (p.value === partner){
      if (p.role === "ADMINISTRATOR" ||
          p.role === "TAC_CHAIR"){
        canWrite = true;
      }
    }
  });
  console.log("Can Wrte: ", canWrite);
  return canWrite;
}

export function canUserWriteTechComments(user, partner){
  let canWrite = false;
  user.partners.forEach( p => {
    if (p.value === partner){
      if (p.role === "ADMINISTRATOR" ||
          p.role === "SALT_ASTRONOMER"){
        canWrite = true;
      }
    }
  });
  return canWrite;
}
