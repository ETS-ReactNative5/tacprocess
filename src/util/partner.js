import _ from "lodash"


export function totalTimeRequestedPerParner(proposals, semester, partner="All" ){
  /*
  *
  * @param proposals of a selected partner all or single
  * @param semester selected semester
  * @param partner all partners or a single
  * @return total
  */

  let total = 0
  proposals.forEach( p =>{
    p.timeRequests.forEach( r => {
      if (r.semester === semester){
        r.distribution.forEach(d => {
          if ( partner === "All"){
            total += d.time
          }else if (d.partnerCode === partner){
              total += d.time
            }
        })

      }
    })
  })
  return total
}

export function userPartners(user) {
  let allPartners = []
  if (! _.isEmpty(user)){
    user.user.role.forEach(u => {
        u.partners.forEach( s => {
          if ( _.findIndex(allPartners, { value: s, label: s }) === -1 ){
            allPartners.push({ value: s, label: s })
          }
        }
      );
    })
  }
  allPartners = [{value: "All", label: "All"}].concat(allPartners)
  return allPartners
}
