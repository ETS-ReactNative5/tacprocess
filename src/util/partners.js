export default class Partner {
    constructor(code, name) {
        this.code = code;
        this.name = name;
    }

    requestsTimeFor(proposal) {
        return proposal.timeRequests.find(t => t.partner.code === this.code && t.time > 0) !== undefined;
    }

    static partnerByCode(code) {
        return PARTNERS.find(partner => partner.code === code);
    }
}

export const PARTNERS = [
    new Partner('AMNH', 'American Museum of Natural History'),
    new Partner('RSA', 'South Africa'),
    new Partner('UW', 'University of Wisconsin')
];
