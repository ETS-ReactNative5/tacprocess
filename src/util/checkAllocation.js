import { isFloat } from "../util";


export function illegalAllocation(element, index, array){
  if(
    !(isFloat(element.allocatedTime.p0)) ||
    !(isFloat(element.allocatedTime.p1)) ||
    !(isFloat(element.allocatedTime.p2)) ||
    !(isFloat(element.allocatedTime.p3)) ||
    !(isFloat(element.allocatedTime.p4)) ||
    !(parseFloat(element.allocatedTime.p0) >= 0) ||
    !(parseFloat(element.allocatedTime.p1) >= 0) ||
    !(parseFloat(element.allocatedTime.p2) >= 0) ||
    !(parseFloat(element.allocatedTime.p3) >= 0) ||
    !(parseFloat(element.allocatedTime.p4) >= 0)
){
    return true
  }
  return false
}

export default function checkAllocatedTimes(proposals){
  /*
   *
   *
   * @param proposals and array of proposals
   * @return true if all the allocated times correct else false
   */
  return !proposals.some(illegalAllocation)
}
