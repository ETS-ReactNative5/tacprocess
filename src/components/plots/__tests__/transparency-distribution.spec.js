import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TransparencyDistribution from './TransparencyDistribution'
import {observingTimeForTransparency} from '../../util'

const proposals = [{
    transparency: 'Clear',
    maxSeeing: 2,
    timeRequests: [
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 150 }, { partnerCode: 'UW', time: 200 }, { partnerCode: 'IUCAA', time: 250 } ]
      },
      {
        semester: '2017-1',
        distribution: [ { partnerCode: 'RSA', time: 400 }, { partnerCode: 'UW', time: 50 }, { partnerCode: 'IUCAA', time: 350 } ]
      }
    ]
  },
  {
    maxSeeing: 2,
    transparency: 'Thin cloud',
    timeRequests: [
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'UW', time: 250 }, { partnerCode: 'RSA', time: 100 }, { partnerCode: 'IUCAA', time: 350 } ]
      },
      {
        semester: '2017-2',
        distribution: [ { partnerCode: 'RSA', time: 800 }, { partnerCode: 'UW', time: 620 } ]
      }
    ]
  },
  {
    transparency: 'Clear',
    maxSeeing: 2,
    timeRequests: [
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'UW', time: 550 }, { partnerCode: 'RSA', time: 600 }, { partnerCode: 'IUCAA', time: 150 } ]
      },
      {
        semester: '2017-1',
        distribution: [ { partnerCode: 'RSA', time: 400 }, { partnerCode: 'UW', time: 320 } ]
      }
    ]
  }
]

// Checking if the TransparencyDistribution Component renders correctly for different input
describe('TransparencyDistribution Component', () => {
  const semester = '2017-1'
  const partner = 'UW'

  // Enzyme method testing if it renders correctly with empty values of the proposals, semester, and partner
  it('Render the TransparencyDistribution Component having unpopulated props with no errors', () => {
    const rendered = mount(<TransparencyDistribution proposals={ [] } semester='' partner=''  />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })

  it('Render the TransparencyDistribution Component having unpopulated props with no errors', () => {
    const rendered = mount(<TransparencyDistribution proposals={ proposals } semester={ semester } partner={ semester }  />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })

})

// Checking the observing time for the TransparencyDistribution
describe('Observing time for the TransparencyDistribution', () => {
  // Testing for a wrong semester
  it('Should return zero for a wrong semester', () => {
    const result = observingTimeForTransparency(proposals, '2019-1', 'Thin cloud')
    expect(result).toBe(0)
  })

  // Testing for a wrong transparency
  it('Should return zero for a wrong transparency', () => {
    const result = observingTimeForTransparency(proposals, '2018-1', 'wrong transparency')
    expect(result).toBe(0)
  })

  // Testing for the correct semester and partner
  it('Should return the correct observing time for the TransparencyDistribution for a semester and partner', () => {
    const result = observingTimeForTransparency(proposals, '2018-1', 'Thin cloud', 'RSA')
    expect(result).toBe(100)
  })
  it('Should return the correct observing time for the TransparencyDistribution for a semester and partner', () => {
    const result = observingTimeForTransparency(proposals, '2018-1', 'Clear', 'UW')
    expect(result).toBe(750)
  })
  it('Should return the correct observing time for the TransparencyDistribution for a semester and partner', () => {
    const result = observingTimeForTransparency(proposals, '2017-1', 'Clear', 'RSA')
    expect(result).toBe(800)
  })

  // Testing for the correct semester and all partners
  it('Should return the correct observing time for the TransparencyDistribution for a semester and all partner', () => {
    const result = observingTimeForTransparency(proposals, '2018-1', 'Thin cloud', 'All')
    expect(result).toBe(700)
    const result1 = observingTimeForTransparency(proposals, '2018-1', 'Thin cloud')
    expect(result1).toBe(700)
  })

})
