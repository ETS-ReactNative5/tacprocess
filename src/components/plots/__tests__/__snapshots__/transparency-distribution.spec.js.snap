// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransparencyDistribution Component Render the TransparencyDistribution Component having unpopulated props with no errors 1`] = `
<TransparencyDistribution
  partner=""
  proposals={Array []}
  semester=""
>
  <Histogram
    datasets={
      Array [
        Object {
          "className": "all-partners",
          "data": Object {
            "Any": 0,
            "Clear": 0,
            "Thick Cloud": 0,
            "Thin cloud": 0,
          },
        },
        Object {
          "className": "partner-only",
          "data": Object {
            "Any": 0,
            "Clear": 0,
            "Thick Cloud": 0,
            "Thin cloud": 0,
          },
        },
      ]
    }
    keys={
      Array [
        "Clear",
        "Thin cloud",
        "Thick Cloud",
        "Any",
      ]
    }
    xLabel="Transparency"
    yLabel="Requested Time (hrs)"
  >
    <div
      className="stat-item"
    >
      <svg
        className="histogram plot"
        height={700}
        width={700}
      />
    </div>
  </Histogram>
</TransparencyDistribution>
`;

exports[`TransparencyDistribution Component Render the TransparencyDistribution Component having unpopulated props with no errors 2`] = `
<TransparencyDistribution
  partner="2017-1"
  proposals={
    Array [
      Object {
        "maxSeeing": 2,
        "timeRequests": Array [
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 150,
              },
              Object {
                "partnerCode": "UW",
                "time": 200,
              },
              Object {
                "partnerCode": "IUCAA",
                "time": 250,
              },
            ],
            "semester": "2018-1",
          },
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 400,
              },
              Object {
                "partnerCode": "UW",
                "time": 50,
              },
              Object {
                "partnerCode": "IUCAA",
                "time": 350,
              },
            ],
            "semester": "2017-1",
          },
        ],
        "transparency": "Clear",
      },
      Object {
        "maxSeeing": 2,
        "timeRequests": Array [
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "UW",
                "time": 250,
              },
              Object {
                "partnerCode": "RSA",
                "time": 100,
              },
              Object {
                "partnerCode": "IUCAA",
                "time": 350,
              },
            ],
            "semester": "2018-1",
          },
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 800,
              },
              Object {
                "partnerCode": "UW",
                "time": 620,
              },
            ],
            "semester": "2017-2",
          },
        ],
        "transparency": "Thin cloud",
      },
      Object {
        "maxSeeing": 2,
        "timeRequests": Array [
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "UW",
                "time": 550,
              },
              Object {
                "partnerCode": "RSA",
                "time": 600,
              },
              Object {
                "partnerCode": "IUCAA",
                "time": 150,
              },
            ],
            "semester": "2018-1",
          },
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 400,
              },
              Object {
                "partnerCode": "UW",
                "time": 320,
              },
            ],
            "semester": "2017-1",
          },
        ],
        "transparency": "Clear",
      },
    ]
  }
  semester="2017-1"
>
  <Histogram
    datasets={
      Array [
        Object {
          "className": "all-partners",
          "data": Object {
            "Any": 0,
            "Clear": 0.4222222222222222,
            "Thick Cloud": 0,
            "Thin cloud": 0,
          },
        },
        Object {
          "className": "partner-only",
          "data": Object {
            "Any": 0,
            "Clear": 0,
            "Thick Cloud": 0,
            "Thin cloud": 0,
          },
        },
      ]
    }
    keys={
      Array [
        "Clear",
        "Thin cloud",
        "Thick Cloud",
        "Any",
      ]
    }
    xLabel="Transparency"
    yLabel="Requested Time (hrs)"
  >
    <div
      className="stat-item"
    >
      <svg
        className="histogram plot"
        height={700}
        width={700}
      />
    </div>
  </Histogram>
</TransparencyDistribution>
`;
